homeassistant/components/philips_js/media_player.py

```diff
@@ -81,70 +85,69 @@ def _inverted(data):
 
 def setup_platform(hass, config, add_entities, discovery_info=None):
     """Set up the Philips TV platform."""
-    name = config.get(CONF_NAME)
-    host = config.get(CONF_HOST)
-    api_version = config.get(CONF_API_VERSION)
-    turn_on_action = config.get(CONF_ON_ACTION)
-
-    tvapi = PhilipsTV(host, api_version)
-    domain = __name__.split(".")[-2]
-    on_script = Script(hass, turn_on_action, name, domain) if turn_on_action else None
-
-    add_entities([PhilipsTVMediaPlayer(tvapi, name, on_script)])
+    hass.async_create_task(
+        hass.config_entries.flow.async_init(
+            DOMAIN,
+            context={"source": config_entries.SOURCE_IMPORT},
+            data=config,
+        )
+    )
+
+
+async def async_setup_entry(
+    hass: HomeAssistantType,
+    config_entry: config_entries.ConfigEntry,
+    async_add_entities,
+):
+    """Set up the configuration entry."""
+    coordinator = hass.data[DOMAIN][config_entry.entry_id]
+    async_add_entities(
+        [
+            PhilipsTVMediaPlayer(
+                coordinator,
+                config_entry.data[CONF_SYSTEM],
+                config_entry.unique_id or config_entry.entry_id,
```

Comment:
Why is there an `or`? All config entries should have unique id, since we set that in this PR and were introducing config entries here.

Reply:
it only have an actual unique_id if we where able to get the serial number from the TV. Not always the case. This is a fallback.

Reply:
We can't use the config entry unique_id api unless the id is actually unique and stable. Otherwise the api can't be trusted.

Reply:
Ok. I think it's unlikely to be problematic here. I'll drop it. Fixed in #46428